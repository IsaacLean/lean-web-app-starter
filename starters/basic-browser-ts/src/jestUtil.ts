/**
 * Clear the root element.
 * This is intended to completely empty the root element's children between tests
 * so test results can be consistent.
 * @param rootEl Optionally pass in the rootEl to be cleared, otherwise a query for the element with the root ID will be made
 */
export const clearRootEl = (rootEl?: HTMLElement) => {
    const r = rootEl ? rootEl : document.getElementById('root')

    if (!r) {
        throw new Error('HTML element with an ID of "root" was not found.')
    }

    r.innerHTML = ''
}

/**
 * Setup the root element.
 * The root div element is intended for other elements to mount to for testing.
 * @returns The new root element
 */
export const setupRootEl = () => {
    const rootEl = document.createElement('div')
    rootEl.setAttribute('id', 'root')
    document.body.replaceChildren(rootEl)

    return rootEl
}
