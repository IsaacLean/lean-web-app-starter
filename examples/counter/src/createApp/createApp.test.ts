import createApp from '.'
import { clearRootEl, setupRootEl } from '../jestUtil'

beforeAll(() => setupRootEl())

afterEach(() => clearRootEl())

test('App element matches snapshot', () => {
    const rootEl = document.getElementById('root')

    if (!rootEl) {
        throw new Error('HTML element with an ID of "root" was not found.')
    }

    expect(createApp(rootEl)).toMatchSnapshot()
})
