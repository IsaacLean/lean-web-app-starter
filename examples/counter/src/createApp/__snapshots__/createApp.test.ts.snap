// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App element matches snapshot 1`] = `
<div
  class="align-items-center d-flex flex-column h-100"
>
  <div
    class="align-items-center container d-flex justify-content-center h-100"
  >
    <div
      class="card"
    >
      <div
        class="card-header"
      >
        Counting App
      </div>
      <div
        class="align-items-center card-body d-flex flex-column px-5"
      >
        <p
          class="lead"
        >
          Count is currently...
        </p>
        <p
          class="display-1"
        >
          0
        </p>
        <div
          class="column-gap-3 d-flex w-100"
        >
          <button
            class="btn btn-success w-50"
          >
            <i
              class="bi bi-dash-circle"
              style="font-size: 1.5rem;"
            >
              <span
                class="sr-only"
              >
                Decrement Count
              </span>
            </i>
          </button>
          <button
            class="btn btn-success w-50"
          >
            <i
              class="bi bi-plus-circle-fill"
              style="font-size: 1.5rem;"
            >
              <span
                class="sr-only"
              >
                Increment Count
              </span>
            </i>
          </button>
        </div>
        <hr
          class="w-100"
        />
        <div
          class="input-group"
        >
          <input
            class="form-control"
            placeholder="Force count to be..."
            type="number"
          />
          <button
            class="btn btn-danger"
            disabled=""
          >
            Force!
          </button>
        </div>
        <hr
          class="w-100"
        />
        <button
          class="btn btn-primary w-100"
        >
          Reset
        </button>
      </div>
    </div>
  </div>
  <footer
    class="text-center text-muted"
    id="footer-text"
  >
    <p
      class="mb-0"
    >
      Learn about the tech stack used for this project and more at its 
      <a
        class="text-muted"
        href="https://github.com/mattlean/lean-js-app-starter/tree/v1.1.0-rc-20241104/examples/counter"
      >
        GitHub repo
      </a>
      .
    </p>
    <p>
      Built with 
      <a
        class="text-muted"
        href="https://github.com/mattlean/lean-js-app-starter"
      >
        Lean JS App Starter
      </a>
      .
    </p>
  </footer>
</div>
`;
